packages/vuetify/src/components/VDataTable/VDataTableHeaderDesktop.ts

```diff
@@ -16,76 +16,95 @@ export default mixins(header).extend({
         },
       }, ['group'])
     },
-    // eslint-disable-next-line max-statements
+    getAria (beingSorted: boolean, isDesc: boolean) {
+      let ariaSort = 'none'
+      let ariaLabel = [
+        this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.sortNone'),
+        this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.activateAscending'),
+      ]
+
+      if (beingSorted) {
+        if (isDesc) {
+          ariaSort = 'descending'
+          ariaLabel = [
+            this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.sortDescending'),
+            this.$vuetify.lang.t(this.options.mustSort
+              ? '$vuetify.dataTable.ariaLabel.activateAscending'
+              : '$vuetify.dataTable.ariaLabel.activateNone'
+            ),
+          ]
+        } else {
+          ariaSort = 'ascending'
+          ariaLabel = [
+            this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.sortAscending'),
+            this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.activateDescending'),
+          ]
+        }
+      }
+
+      return { ariaSort, ariaLabel: ariaLabel.join(' ') }
+    },
     genHeader (header: DataTableHeader) {
-      const listeners: any = {}
-      const children = []
-      const attrs = {
-        role: 'columnheader',
-        scope: 'col',
-        'aria-label': header.text || '',
-        'aria-sort': 'none',
+      const data: Required<Pick<VNodeData, 'attrs' | 'on' | 'class' | 'style'>> = {
+        attrs: {
+          role: 'columnheader',
+          scope: 'col',
+          'aria-label': header.text || '',
```

Comment:
I think that should have a fallback of some sort but I'm not sure exactly how. @phiter  should we ever provide an empty aria-label?

Reply:
You don't really need `aria-label` on a `columnheader` if the column already has the same text as a text node.

Reply:
We only need `aria-label` when the element does not have a text (if it only has an icon, for example) or if we want to provide a different text for the screen reader.